hive-metastore/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java

```diff
@@ -60,8 +66,23 @@ public class HiveCreateReplaceTableTest extends HiveMetastoreTest {
 
   private String tableLocation;
 
+  @RegisterExtension
+  public static final HiveMetastoreExtension HIVE_METASTORE_EXTENSION =
```

Comment:
this should be private

Reply:
I made public as per [reference](https://junit.org/junit5/docs/5.1.1/api/org/junit/jupiter/api/extension/RegisterExtension.html)

Reply:
The reference you pointed to defines the `RegisterExtension` interface via `public @interface RegisterExtension` but that doesn't mean we need to make it public here where it is used. That being said, it should be defined `private` here and in all other places (similar to how it's defined in the `TestHiveCatalog` from your other PR